// sprites/index.js
    exports.SpriteArray=[];
    exports.CharacterArray=[];
    exports.PlatformArray=[];
    exports.LaserArray=[];
exports.pictures=["skitty","spearow","piplup"];

var pixels=20;
    
    var Sprite=class {
        constructor(user,picture,x,y,falling){
            this.user=user,
            this.picture=picture,
            this.x=x*pixels,
            this.y=y*pixels,
            this.left=true, //if left is true, use the left lloking sprite, if false use the right one
            this.width=pixels*4,
            this.height=pixels*4,
            this.falling=falling;
       //     exports.SpriteArray.push(this);
            }
        
    };
    class Character extends Sprite {
        constructor(user,picture,x,y,falling){
            super(user,picture,x,y);
            this.user=user,
            this.left=true;
            this.width=pixels*3,
            this.height=pixels*3,
            this.falling=falling; //if they're not falling, they're jumping
            this.health=10;
            this.energy=100;
            this.notCharacter=false;
            this.jumpNum=16;
            this.resetJump=function(){
              this.jumpNum=16;  
            };
           // SpriteArray.push(this);
            exports.CharacterArray.push(this);
        }
    }
   class Platform extends Sprite {
        constructor(user,picture,x,y,falling,width,height){
            super(user,picture,x,y);
            this.falling=false; //platforms just kinda hover
            this.width=width*pixels,
            this.height=height*pixels,
            this.notCharacter=true;
          //  SpriteArray.push(this);
            exports.PlatformArray.push(this);
        }
    }

    class Laser extends Sprite {
        constructor(user,picture,x,y,falling,left){
            super(user,picture,x,y);
            this.width=pixels*1,
            this.height=pixels*1,
            this.falling=false;
            this.left=left;
            this.notCharacter=true;
         //   exports.SpriteArray.push(this);
            exports.LaserArray.push(this);
            console.log("laser made!");
        }
    }


exports.SpriteArray.push(new Platform("server","block", 2.5,15,false,9,1));
exports.SpriteArray.push(new Platform("server","block", 11.25, 12.5,false,8,1));
exports.SpriteArray.push(new Platform("server","block", 35, 22.5,false,5,1));
exports.SpriteArray.push(new Platform("server","block", 28, 18.5,false,5,1));
exports.SpriteArray.push(new Platform("server","block", 35, 14.5,false,5,1));
exports.SpriteArray.push(new Platform("server","block", 28, 10.5,false,5,1));
exports.SpriteArray.push(new Platform("server","block", 35, 6.5,false,5,1));
exports.SpriteArray.push(new Platform("server","block", 7.5,22.5,false,5,1));
exports.SpriteArray.push(new Platform("server","floor",-2.5,25,false,50,5));

exports.Sprite=Sprite;
exports.Laser=Laser;
exports.Character=Character;
exports.Platform=Platform;
exports.pixels=pixels;
